src/main/java/de/idealo/spring/stream/binder/sns/health/SnsBinderHealthIndicator.java

```diff
@@ -0,0 +1,67 @@
+package de.idealo.spring.stream.binder.sns.health;
+
+import static java.util.stream.Collectors.toList;
+
+import java.util.Collections;
+import java.util.List;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.boot.actuate.health.AbstractHealthIndicator;
+import org.springframework.boot.actuate.health.Health;
+import org.springframework.cloud.stream.config.BindingServiceProperties;
+import org.springframework.util.Assert;
+
+import com.amazonaws.SdkClientException;
+import com.amazonaws.services.sns.model.ListTopicsResult;
+
+import de.idealo.spring.stream.binder.sns.SnsMessageHandlerBinder;
+
+public class SnsBinderHealthIndicator extends AbstractHealthIndicator {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(SnsBinderHealthIndicator.class);
+
+    private final SnsMessageHandlerBinder snsMessageHandlerBinder;
+    private final BindingServiceProperties bindingServiceProperties;
+
+    public SnsBinderHealthIndicator(final SnsMessageHandlerBinder snsMessageHandlerBinder, final BindingServiceProperties bindingServiceProperties) {
+        Assert.notNull(snsMessageHandlerBinder, "SnsMessageHandlerBinder must not be null");
+        this.snsMessageHandlerBinder = snsMessageHandlerBinder;
+        Assert.notNull(bindingServiceProperties, "BindingServiceProperties must not be null");
+        this.bindingServiceProperties = bindingServiceProperties;
+    }
+
+    @Override
+    protected void doHealthCheck(Health.Builder builder) {
+        boolean allListenersRunning = true;
```

Comment:
I suppose you don't need this variable anymore? Since you have a simple check method you can just use if+else or a return.